sdk/core/System.ClientModel/samples/ServiceMethods.md

```diff
@@ -0,0 +1,117 @@
+
+```C# Snippet:ReadmeGetRawResponse
+// Access the HTTP response details.
+PipelineResponse response = result.GetRawResponse();
+
+Console.WriteLine($"Response status code: '{response.Status}'.");
+Console.WriteLine("Response headers:");
+foreach (KeyValuePair<string, string> header in response.Headers)
+{
+    Console.WriteLine($"Name: '{header.Key}', Value: '{header.Value}'.");
+}
+```
+
+## Protocol methods
+
+In contrast to convenience methods, **protocol methods** are service methods that provide very little convenience over the raw HTTP APIs a cloud service exposes. They represent request and response message bodies using types that are very thin layers over raw JSON/binary/other formats. Users of client protocol methods must reference a service's API documentation directly, rather than relying on the client to provide developer conveniences via strongly-typing service schemas.
+
+The following sample illustrates how to call a protocol method, including creating the request payload and accessing the details of the HTTP response.
+
+```C# Snippet:ServiceMethodsProtocolMethod
+// Create a BinaryData instance from a JSON string literal.
+BinaryData input = BinaryData.FromString("""
+    {
+        "countryRegion": {
+            "isoCode": "US"
+        },
+    }
+    """);
+
+// Create a BinaryContent instance to set as the HTTP request content.
+BinaryContent requestContent = BinaryContent.Create(input);
+
+// Call the protocol method
+ClientResult result = await client.AddCountryCodeAsync(requestContent);
+
+// Obtain the output response content from the returned ClientResult.
+BinaryData output = result.GetRawResponse().Content;
+
+using JsonDocument outputAsJson = JsonDocument.Parse(output.ToString());
+string isoCode = outputAsJson.RootElement
+    .GetProperty("countryRegion")
+    .GetProperty("isoCode")
+    .GetString();
+
+Console.WriteLine($"Code for added country is '{isoCode}'.");
+```
+
+Protocol methods take an optional `RequestOptions` parameter.  `RequestOptions` can be used to modify various aspects of the HTTP request sent by the service method, such as adding a request header, or adding a policy to the client pipeline that can modify the request directly before sending it to the service.  `RequestOptions` also enables passing a `CancellationToken` to the method.
+
+```C# Snippet:RequestOptionsReadme
+// Create RequestOptions instance.
+RequestOptions options = new();
+
+// Set the CancellationToken.
+options.CancellationToken = cancellationToken;
+
+// Add a header to the request.
+options.AddHeader("CustomHeader", "CustomHeaderValue");
+
+// Create an instance of a model that implements the IJsonModel<T> interface.
+CountryRegion region = new("US");
+
+// Create BinaryContent from the input model.
+BinaryContent content = BinaryContent.Create(region);
+
+// Call the protocol method, passing the content and options.
+ClientResult output = await client.AddCountryCodeAsync(content, options);
+```
+
```

Comment:
Suggestion: please put a note somewhere describing how a client operator method applies the RequestOptions passed in as a method argument. Give a link to PipleineMessage.Apply()

Reply:
Added to the more detailed samples page on client implementation.  Let me know if you have further suggestions, and thanks @dargilco!